src/osha/oira/client/browser/conditions.py

```diff
@@ -10,17 +11,16 @@ def terms_changed(self):
         return getattr(self.account, "tc_approved", None) is not None
 
     def __call__(self):
-        self.came_from = self.request.form.get("came_from")
-        if isinstance(self.came_from, list):
-            # If came_from is both in the querystring and the form data
-            self.came_from = self.came_from[0]
+        webhelpers = api.content.get_view(
+            name="webhelpers", context=self.context, request=self.request
+        )
+        self.came_from = webhelpers.get_came_from(
```

Comment:
I do not like and understand what is this passion in using instance attributes instead of plain variables...

Reply:
I don't know what passion you mean. I didn't change anything about the `came_from` attribute. FWIW, in general, instance attributes are accessible from e.g. the template while local (is that what you mean by “plain”?) variables aren't.